Ignore certain external customer domains

joker-at-work opened the pull request:
We didn't get time to properly do this so we now add a hard-coded list of domain names via config that are not supposed to be treated like external customer domains even though they totally look like they should be.

I hope we can revert this and build something proper with feature-flags on domains.

Change-Id: I2a917a027d3416cf9bbe80e766b912ceed650bc3
Related-Change-Id: I034e89251ce44310f6642983323bed26eddc0159
Related-Change-Id: Iea2cb2c1726adb4a4b6f6b2d6cf0875972f4b517
Related-Change-Id: Ia065958822b636c35b84349f44e88f38c923acbb

joker-at-work committed:
Ignore certain external customer domains

We didn't get time to properly do this so we now add a hard-coded list
of domain names via config that are not supposed to be treated like
external customer domains even though they totally look like they should
be.

I hope we can revert this and build something proper with feature-flags
on domains.

Change-Id: I2a917a027d3416cf9bbe80e766b912ceed650bc3
Related-Change-Id: I034e89251ce44310f6642983323bed26eddc0159
Related-Change-Id: Iea2cb2c1726adb4a4b6f6b2d6cf0875972f4b517
Related-Change-Id: Ia065958822b636c35b84349f44e88f38c923acbb

grandchild reviewed (dismissed):
LGTM

And just to check: `nova/virt/vmwareapi/driver.py:631`ff doesn't need changing because we want all those prefixes as traits even if some of them are actually excluded again, right?

joker-at-work commented:
> doesn't need changing because we want all those prefixes as traits even if some of them are actually excluded again, right?

yes. the ignore-list is just for ignoring _some_ of the domains matched by a prefix. in general, the prefixes are all valid.

dhoeller reviewed (dismissed):
LGTM

joker-at-work committed:
vmware: Fix external customer DRS group sync

When syncing the external customer DRS groups during spawning of a VM,
we used to take the customer's context to query the DB for instances.
This lead to us removing other projects' VMs from the DRS VM groups
until the next sync-loop. To fix this, we elevate the context to an
admin context which allows us to query all projects' instances.

Note:
While server-groups look like having a similar problem at first, they
are only valid on project-level and thus only contain instances
belonging to the same project as the currently-handled VM.

Change-Id: Id4cf263f6699b3d3acee9d767852173f47d45811
Fixup-For: Ia065958822b636c35b84349f44e88f38c923acbb

dhoeller reviewed (approved):
LGTM
